### PR TITLE
Update port renumbering logic for constants

### PR DESCRIPTION
**Pull request by achetver**
Description: Fixed incorrect port numbering for constant operations in IR.

JIRA: 42465

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests: N/A - No new implemented functions;
* [x]  Framework operation tests: N/A - No new enabled operations;
* [x]  Transformation tests: N/A - No new implemented transformations;
* [x]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py: N/A - No affected changes.
* [x]  Model Optimizer IR Reader check: Done manually

Documentation:
* [x]  Supported frameworks operations list: N/A - No new supported operations;
* [x]  Supported **public** models list: N/A - Nothing to add;
* [x]  New operations specification: N/A - No new supported operations;
* [x]  Guide on how to convert the **public** model: N/A - Nothing to add;
* [x]  User guide update: N/A - Nothing to update.

### CODE REVIEW

File: model-optimizer/mo/back/ie_ir_ver_2/emitter.py [hunk 1]
```diff
@@ -440,9 +440,10 @@ def port_renumber(graph: Graph):
         node = Node(graph, node)
         if node.kind == 'op':
```
Comment (lazarevevgeny) [inline model-optimizer/mo/back/ie_ir_ver_2/emitter.py:441]: nit: Let's refactor lines 439-441 with just one line: `for node in graph.get_op_nodes():`

File: model-optimizer/mo/utils/ir_reader/layer_to_class.py [hunk 2]
```diff
@@ -123,8 +123,9 @@ def restore_correct_ports(graph: Graph):
             in_port_id = d['in'] if not is_control_flow else 'control_flow_' + str(d['in'])
             to_node_attrs['_in_ports'].update({in_port_id: {'control_flow': is_control_flow}})
         if 'out' in d:
-            num_of_in_nodes = len(Node(graph, u).in_nodes())
-            decremented_number = d['out'] - num_of_in_nodes
+            node = Node(graph, u)
```
Comment (lazarevevgeny) [inline model-optimizer/mo/utils/ir_reader/layer_to_class.py:126]: Please, add comments here why `Const` op is handled specially

File: model-optimizer/mo/back/ie_ir_ver_2/emitter.py [hunk 3]
```diff
@@ -161,7 +161,7 @@ def xml_ports(node: Node, element: Element, edges: Element):
                 outputs = SubElement(element, 'output')
             port = SubElement(outputs, 'port')
             port.set('id', str(d['out']))
-            port_id = d['out'] - len(node.in_nodes())
+            port_id = d['out'] - len(node.in_nodes()) if node.type != 'Const' else d['out']
```
Comment (lazarevevgeny) [inline model-optimizer/mo/back/ie_ir_ver_2/emitter.py:164]: This solution works but it does not look beautiful. Can we consider refactoring of Const op IR generation so there will be no need in this check here?

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

File: model-optimizer/mo/back/ie_ir_ver_2/emitter.py (hunk 1 above)
Comment (achetver) [reply model-optimizer/mo/back/ie_ir_ver_2/emitter.py]: Done.

File: model-optimizer/mo/utils/ir_reader/layer_to_class.py (hunk 2 above)
Comment (achetver) [reply model-optimizer/mo/utils/ir_reader/layer_to_class.py:126]: Done.

File: model-optimizer/mo/back/ie_ir_ver_2/emitter.py (hunk 3 above)
Comment (achetver) [reply model-optimizer/mo/back/ie_ir_ver_2/emitter.py:166]: To be honest I don't know how to make this part better. I have tried some variants, but they are looks bad.

Comment (lazarevevgeny) [reply model-optimizer/mo/back/ie_ir_ver_2/emitter.py:166]: The transformation `CreateConstNodesReplacement` from file `model-optimizer/extensions/back/SpecialNodesFinalization.py` adds a copy of the data node in lines 104-106 as input to the `Const` op node. This data node was added to properly generate the Const operation long time ago (this data node is the reason why the Const output port start from 1 and not 0). I would recommend to not add this redundant data node and modify the emitter code to correctly work for this case.

Comment (lazarevevgeny) [reply model-optimizer/mo/back/ie_ir_ver_2/emitter.py:166]: Never mind. Please, ignore this comment.

**Review by lazarevevgeny — APPROVED** [review]

**Pull request merged**